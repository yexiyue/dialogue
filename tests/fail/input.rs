use dialogue_core::Asker;

#[derive(Asker,Debug)]
struct User {
    #[abc]
    name: String,
    #[input]
    email: String,
    #[input(prompt = "请输入邮箱2")]
    email2: Option<String>,
}

fn main() {
    let mut user = User::asker();
    let user = user.email2().name().email().finish();
    println!("{:?}", user);
}
