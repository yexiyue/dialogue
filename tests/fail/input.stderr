error: expected attribute arguments in parentheses: #[input(...)]
 --> tests/fail/input.rs:7:7
  |
7 |     #[input]
  |       ^^^^^

error: cannot find attribute `abc` in this scope
 --> tests/fail/input.rs:5:7
  |
5 |     #[abc]
  |       ^^^

error[E0599]: no function or associated item named `asker` found for struct `User` in the current scope
  --> tests/fail/input.rs:14:26
   |
4  | struct User {
   | ----------- function or associated item `asker` not found for this struct
...
14 |     let mut user = User::asker();
   |                          ^^^^^ function or associated item not found in `User`
